### PR TITLE
[music] align template and add test

### PR DESCRIPTION
**Pull request by xhr0804**
### Purpose
*What is the context of this pull request? Why is it being done?*

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: skills/csharp/experimental/musicskill/Adapters/DefaultAdapter.cs [hunk 1]
```diff
@@ -1,60 +1,124 @@
...
+        }
+
+        private async Task SendErrorMessageAsync(ITurnContext turnContext, Exception exception)
+        {
+            try
+            {
+                _telemetryClient.TrackException(exception);
+
+                // Send a message to the user.
+                await turnContext.SendActivityAsync(_templateEngine.GenerateActivityForLocale("ErrorMessage"));
```
Comment (bobokids) [inline skills/csharp/experimental/musicskill/Adapters/DefaultAdapter.cs]: SharedResponses.ErrorMessage

File: skills/csharp/experimental/musicskill/Dialogs/MainDialog.cs [hunk 2]
```diff
@@ -23,21 +23,20 @@ namespace MusicSkill.Dialogs
...
-            IServiceProvider serviceProvider,
-            IBotTelemetryClient telemetryClient)
+            IServiceProvider serviceProvider)
             : base(nameof(MainDialog))
         {
             _settings = serviceProvider.GetService<BotSettings>();
             _services = serviceProvider.GetService<BotServices>();
             _templateManager = serviceProvider.GetService<LocaleTemplateManager>();
-            TelemetryClient = telemetryClient;
+            TelemetryClient = TelemetryClient;
```
Comment (bobokids) [inline skills/csharp/experimental/musicskill/Dialogs/MainDialog.cs]: remove

**Comment by bobokids** [comment]
>         protected BotSettings Settings { get; set; }

remove set

---
Refers to: skills/csharp/experimental/musicskill/Dialogs/SkillDialogBase.cs:50 in fd9f6c3. [](commit_id = fd9f6c3ea22aaf0ad72a3ce559e0981922901d4a, deletion_comment = False)

File: skills/csharp/tests/musicskill.Tests/Utterances/GeneralUtterances.Designer.cs [hunk 3]
```diff
@@ -0,0 +1,207 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     此代码由工具生成。
```
Comment (bobokids) [inline skills/csharp/tests/musicskill.Tests/Utterances/GeneralUtterances.Designer.cs]: please check the file header.

File: skills/csharp/experimental/musicskill/Adapters/DefaultAdapter.cs (hunk 1 above)
Comment (xhr0804) [reply skills/csharp/experimental/musicskill/Adapters/DefaultAdapter.cs]: fixed

File: skills/csharp/tests/musicskill.Tests/Utterances/GeneralUtterances.Designer.cs (hunk 3 above)
Comment (xhr0804) [reply skills/csharp/tests/musicskill.Tests/Utterances/GeneralUtterances.Designer.cs]: fixed

File: skills/csharp/experimental/musicskill/Startup.cs [hunk 4]
```diff
@@ -118,6 +119,12 @@ public void ConfigureServices(IServiceCollection services)
 
             services.AddSingleton(new LocaleTemplateManager(localizedTemplates, settings.DefaultLocale ?? "en-us"));
 
+            // Configure service manager
+            services.AddSingleton<IServiceManager>(new ServiceManager(settings));
+
+            // Register AuthConfiguration to enable custom claim validation.
+            services.AddSingleton(sp => new AuthenticationConfiguration { ClaimsValidator = new AllowedCallersClaimsValidator(sp.GetService<IConfiguration>()) });
+
```
Comment (xieofxie) [inline skills/csharp/experimental/musicskill/Startup.cs]: need the following after `.UseEndpoints(endpoints => endpoints.MapControllers());`
```
            // Uncomment this to support HTTPS.
            // app.UseHttpsRedirection();
```

File: skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        public async Task Test_Sample_Dialog_SkillMode()
+        {
+            await GetSkillTestFlow()
+               .Send(PlayMusicDialogUtterances.PlayMusic)
+                .AssertReply((activity) =>
+                {
+                    var a = (Activity)activity;
+                    Assert.AreEqual(ActivityTypes.Event, a.Type);
+                    Assert.AreEqual(typeof(OpenDefaultApp), a.Value.GetType());
+                    Assert.AreEqual((a.Value as OpenDefaultApp).MusicUri, "spotify:playlist:37i9dQZF1DXcCnTAt8Cf");
```
Comment (ssss141414) [inline skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs]: Put this "spotify:playlist:37i9dQZF1DXcCnTAt8Cf" to a field? Bacause it's used many times and is a little ugly hardcode here.

File: skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                   var a = (Activity)activity;
+                   Assert.AreEqual(ActivityTypes.EndOfConversation, a.Type);
+                   Assert.AreEqual(typeof(ActionResult), a.Value.GetType());
+                   Assert.AreEqual((a.Value as ActionResult).ActionSuccess, false);
+               })
+               .StartTestAsync();
+        }
+
+        [TestMethod]
+        public async Task Test_Sample_Action_w_Input()
```
Comment (xieofxie) [inline skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs]: what is w?

File: skills/csharp/experimental/musicskill/Startup.cs (hunk 4 above)
Comment (xhr0804) [reply skills/csharp/experimental/musicskill/Startup.cs]: added

File: skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs (hunk 5 above)
Comment (xhr0804) [reply skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs]: sure, moved

File: skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs (hunk 6 above)
Comment (xhr0804) [reply skills/csharp/tests/musicskill.Tests/PlayMusicDialogTests.cs]: this is just a shorthand for "with" as Sample Skill did. But anyway, changed it.

**Pull request merged**